drivers/SmartThings/zigbee-switch/src/zigbee-dual-metering-switch/init.lua

```diff
@@ -25,31 +32,91 @@ local function can_handle_zigbee_dual_metering_switch(opts, driver, device, ...)
   return false
 end
 
-local function endpoint_to_component(device, endpoint)
-  if endpoint == 2 then
-    return "switch1"
-  else
-    return "main"
+local function device_added(driver, device, event)
+  if device.network_type == st_device.NETWORK_TYPE_ZIGBEE then
+    for i = 1,1 do
+      local name = "AURORA Outlet 2"
+      local metadata = {
+        type = "EDGE_CHILD",
+        label = name,
+        profile = "switch-power-2",
+        parent_device_id = device.id,
+        parent_assigned_child_key = string.format("%02X", i),
+        vendor_provided_label = name,
+      }
+      driver:try_create_device(metadata)
+    end
   end
 end
 
-local function component_to_endpoint(device, component)
-  if component == "switch1" then
-    return 2
-  else
-    return 1
+local function find_child(parent, ep_id)
+  return parent:get_child_by_parent_assigned_key(string.format("%02X", ep_id))
+end
+
+local function device_init(driver, device)
+  if device.network_type == st_device.NETWORK_TYPE_ZIGBEE then
+    device:set_find_child(find_child)
   end
 end
 
-local function map_components(self, device)
-  device:set_endpoint_to_component_fn(endpoint_to_component)
-  device:set_component_to_endpoint_fn(component_to_endpoint)
+local function switch_on_command_handler(driver, device, command)
+  device:send(OnOff.server.commands.On(device))
+  
+end
+
+local switch_off_command_handler = function(driver, device, command)
+  device:send(OnOff.server.commands.Off(device))
+end
+
+local function on_off_command_handler(driver, device, value, zb_rx)
+  local event
+  
+  if value == OnOff.server.commands.On.ID then
+    event = capabilities.switch.switch.on()
+    device:set_field(CURRENT_STATUS, "on")
```

Comment:
where are these used?

Reply:
that is outdated piece of code, was not needed :)